state-sync/src/coordinator.rs

```diff
@@ -1631,15 +1630,16 @@ impl<T: ExecutorProxyTrait> StateSyncCoordinator<T> {
         });
 
         ready.into_iter().for_each(|(peer, request_info)| {
-            let result_label =
-                if let Err(err) = self.deliver_subscription(peer.clone(), request_info) {
-                    error!(LogSchema::new(LogEntry::SubscriptionDeliveryFail)
-                        .peer(&peer)
-                        .error(&err));
-                    counters::FAIL_LABEL
-                } else {
-                    counters::SUCCESS_LABEL
-                };
+            let result_label = if let Err(err) =
+                self.deliver_subscription(peer.clone(), request_info, highest_li_version)
+            {
+                error!(LogSchema::new(LogEntry::SubscriptionDeliveryFail)
+                    .peer(&peer)
+                    .error(&err));
+                counters::FAIL_LABEL
+            } else {
+                counters::SUCCESS_LABEL
```

Comment:
We need to make another pass later between you and me and cleanup this weird anti-pattern in the metrics in Mempool / State Sync.  Passing labels around makes it really hard to understand what's going on.